### DISCUSSION TITLE
Lua global environments

### DISCUSSION THREAD

**Discussion by Whoome** [neolithos/neolua#145] [Q&A] [answered]
I'm trying to figure out if there is a parallel to the Lua engine's ability to replace _G in the lua environment.  (And then perhaps switch it back out).

What I have is a situation where I create a protected environment table in lua, then swap that in for _G so that everything now runs in that context.  But... I can't see how I would update the global table or set the global table from the current API.  

I suspect I am missing something.

**Comment by Whoome** [comment, marked as answer]
What I am attempting now is using the env parameter from the LuaChunk.Run.  I -think- this does what I want.

**Reply by neolithos** [reply]
Correct, env is mapped to _G.
